src/Storages/ObjectStorage/DataLakes/Iceberg/IcebergMetadata.cpp

```diff
@@ -970,13 +1039,27 @@ ColumnMapperPtr IcebergMetadata::getColumnMapperForObject(ObjectInfoPtr object_i
     return persistent_components.schema_processor->getColumnMapperById(iceberg_object_info->underlying_format_read_schema_id);
 }
 
-ColumnMapperPtr IcebergMetadata::getColumnMapperForCurrentSchema() const
+ColumnMapperPtr IcebergMetadata::getColumnMapperForCurrentSchema(StorageSnapshotPtr storage_snapshot, ContextPtr context) const
 {
+    auto iceberg_table_state = extractIcebergSnapshotIdFromMetadataObject(storage_snapshot);
+    // This is a temporary cludge for cluster functions because now the state on replicas is not synchronized with the primary state on the coordinator
+    if (!iceberg_table_state)
+    {
+        auto configuration_ptr = configuration.lock();
+        if (!configuration_ptr)
```

Comment:
Let's move initialization of `configuration_ptr` before the `if (!iceberg_table_state)`, because it is also used rigth after that `if`